src/main/java/org/kitteh/irc/client/library/implementation/NettyManager.java

```diff
@@ -272,10 +273,27 @@ private void schedule(boolean force) {
             }
         }
 
+        private void shutdown(@Nullable String message, boolean reconnect, boolean fromTimeout) {
+            if (fromTimeout && message == null) {
+                shutdown(this.client.getInteractionDefaults().getDefault(Interaction.QUIT_PING_TIMEOUT), reconnect);
+            }
+
+            shutdown(message, reconnect);
+        }
+
         private void shutdown(@Nullable String message, boolean reconnect) {
             this.reconnect = reconnect;
 
-            this.sendMessage("QUIT" + ((message != null) ? (" :" + message) : ""), true);
+            String shutdownMessage = message;
+            if (message == null) {
```

Comment:
You are losing the ability to send null in `Client#shutdown(String)` this way. Oops!